asyncpg/connection.py

```diff
@@ -55,15 +57,40 @@ def __init__(self, protocol, transport, loop, addr, opts, *,
 
         self._listeners = {}
 
-        ver_string = self._protocol.get_settings().server_version
+        settings = self._protocol.get_settings()
+        ver_string = settings.server_version
         self._server_version = \
             serverversion.split_server_version_string(ver_string)
 
+        self._server_caps = caps = capabilities.detect_server_capabilities(
+            self._server_version, settings)
+
         if self._server_version < (9, 2):
             self._intro_query = introspection.INTRO_LOOKUP_TYPES_91
         else:
             self._intro_query = introspection.INTRO_LOOKUP_TYPES
 
+        _reset_query = ''
```

Comment:
Can you factor this code into a separate method `_get_reset_query()`? I'd also only call it from `Connection.reset()` (only pooled connections are likely to ever call `reset()`).